sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -341,6 +341,12 @@ case class LoadDataCommand(
       } else {
         val uri = new URI(path)
         if (uri.getScheme() != null && uri.getAuthority() != null) {
+          val hadoopConf = sparkSession.sessionState.newHadoopConf()
+          val srcPath = new Path(path)
```

Comment:
nit: Let's use `new Path(uri)`. I think we better validate `uri` in this scope.

Reply:
But will this lose the normalization in `new Path(path)` ? Or normalization in `URI` covers it?

Reply:
I think here `path` is an URI here otherwise `val uri = new URI(path)` should fail, and we return `uri`. So, I think we should check if the `uri` is valid.